### PR TITLE
Feature/auto update

### PR DESCRIPTION
**Pull request by grantdfoster**

### CODE REVIEW

File: types/auth.ts [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 }
 
 export interface RegisterCredentials extends LoginCredentials {
-    passwordConfirmation: string,
-    name: string | null
-}
\ No newline at end of file
+  passwordConfirmation: string;
+  name: string | null;
+}
```
Comment (mpothier) [inline types/auth.ts]: We should see if it's possible to share one types directory for both the electron app and the Firebase Functions. I was able to do this with a regular web app, but I'm not sure what might happen with Electron when it builds and compiles.

Comment (mpothier) [reply types/auth.ts]: (Currently the Functions have a separate, duplicate `types` files)

File: src/App.vue [hunk 2]
```diff
@@ -33,15 +33,15 @@ export default Vue.extend({
...
   }
 });
 </script>
 <style lang="scss" scoped>
-#router {
-  margin-left: 55px;
-}
+// #router {
+//   margin-left: 55px;
+// }
```
Comment (mpothier) [inline src/App.vue:44]: Maybe this should be conditional on whether or not the sidebar is active (i.e. logged in or not?)

**Review by mpothier — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "name": "package-manager",
   "description": "does things",
   "author": "AEC Lab",
-  "version": "0.0.3",
+  "version": "0.0.4",
```
[the remark this answers is not in the document]
Comment (grantdfoster) [reply package.json:5]: We should definitely come up with one.  Right now I am minimally bumping the version every time we anticipate a master commit (so CI releases work).  There are solid conventions around this though and we should follow those for now - https://semver.org/

File: src/background.ts [hunk 4]
```diff
@@ -1,116 +1,168 @@
 "use strict";
 
 import { app, protocol, BrowserWindow } from "electron";
+import { autoUpdater } from "electron-updater";
```
[the remark this answers is not in the document]
Comment (grantdfoster) [reply src/background.ts:4]: Ah!  I believe this library is just to look at new versions of _our_ app, not updating Electron itself.  Our package.json should ensure Electron (and other package versions) stay the same.  That being said, compatibility with electron and other libraries is historically finicky so we'll keep an eye out

File: src/components/Menu.vue [hunk 5]
```diff
@@ -50,52 +45,57 @@
...
     if (loggedOut) {
-      const r = this.$router.resolve({
-          path: this.$route.path
-      });
-      window.location.assign(r.href)
+      // const r = this.$router.resolve({
+      //   path: this.$route.path
+      // });
+      // window.location.assign(r.href);
+      this.route("login");
```
[the remark this answers is not in the document]
Comment (grantdfoster) [reply src/components/Menu.vue:89]: Makes sense - yeah I believe this was resolved w/ that new logic... I tested email, Google, and GitHub logins / logouts yesterday!

File: types/auth.ts (hunk 1 above)
Comment (grantdfoster) [reply types/auth.ts]: Good catch, will definitely look into this / create an issue for it.  We can also place those Vue `.shim` files in a `types` directory too.

File: src/App.vue (hunk 2 above)
Comment (grantdfoster) [reply src/App.vue:44]: Yeah agreed, I just wanted your login dialog to be centered for this commit haha.  I haven't looked deep into the Vuetify menu API, I'm sure there is a smart way to handle this in general

**Pull request merged**